java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -446,7 +448,8 @@ abstract class AbstractArcHost(
         existingParticleContext: ParticleContext?,
         context: ArcHostContext
     ): ParticleContext {
-        val particle = instantiateParticle(ParticleIdentifier.from(spec.location), spec)
+        val id = if (spec.isExpressionParticle()) EvaluatorParticle::class.toParticleIdentifier() else ParticleIdentifier.from(spec.location)
```

Comment:
I feel like we could avoid special casing this by pre-registering it in constructor of AbstractArcHost. 

That is, why can't we modify the Plan.Particle ctor() so that if you put 'null' for the location and it's an expression particle, we default it to EvaluatorParticle?

Then everything would just "work" and if you wanted, the recipe2plan generator could just act like you wrote
particle EvaluatorParticle in 'arcs.core.host.EvaluatorParticle' as syntactic sugar?

Reply:
Yup, reverted, no need for special casing - we will generate the explicit location in the recipe2plan.